neurons/validator.py

```diff
@@ -44,12 +46,68 @@ class Validator(BaseValidatorNeuron):
 
     def __init__(self, config=None):
         super(Validator, self).__init__(config=config)
-        # competition_id to (hotkey_uid, days_as_leader)
-        self.competitions_leaders = {}
-        self.load_state()
+        self.scheduler_config = config_for_scheduler(self.config, self.hotkeys, test_mode=True)
+        self.rewarder = Rewarder(self.rewarder_config)
         
-    async def forward(self):
-        ...
+        asyncio.run_coroutine_threadsafe(self.competition_loop(self.scheduler_config, self.rewarder_config), self.loop)
+
+    async def competition_loop(self, scheduler_config: dict[str, CompetitionManager], rewarder_config: RewarderConfig):
+        """Example of scheduling coroutine"""
+        bt.logging.info("Running Competition Loop.")
+        while True:
+            competition_result = await run_competitions_tick(scheduler_config)
+            bt.logging.debug(f"Competition result: {competition_result}")
+            if competition_result:
+                winning_evaluation_hotkey, competition_id = competition_result
+
+                # update the scores
+                await self.rewarder.update_scores(winning_evaluation_hotkey, competition_id)
+                self.rewarder_config = RewarderConfig(self.rewarder.competition_leader_mapping, self.rewarder.scores)
```

Comment:
rewarderconfig doesn't fit here

Reply:
we store rewarder config locally and it needs to be updated after update_scores is run
isnt updating it right away the most safe way?